### PR TITLE
Support for copying between Armatures with different Scale

### PR DESCRIPTION
**Pull request by Float3**
Added Scale Adjustment for copying DynamicBones, DynamicBoneColliders, and regular Colliders between Armatures with different scales.

This assumes uniform scaling on all axis as only the x component of both the destination transform and the source transform is compared
```
public static float GetScaleMultiplier(Transform from, Transform to)
        {
            return from.lossyScale.x / to.lossyScale.x;
        }
```

this addresses Issue #11

### CODE REVIEW

**Comment by Float3** [comment]
Possible (Known) Issues/Limitations: 
-Adjusting for the scale difference on a collider requires changing the local position of a the GameObject that the collider is on. If that GameObject happens to be part of the HumanoidRig it will cause issues for obvious reasons. Fortunately, it's common to have colliders on separate GameObjects

- non-uniform scale difference is not supported. e.g.
```
sourcescale = Vector3(1,1,1);
destinationscale = Vector3(100,50,100);
```

File: Scripts/Editor/Copiers/LegacyCopier.cs [hunk 1]
```diff
@@ -369,6 +395,28 @@ internal static void CopyAllColliders(GameObject from, GameObject to, bool creat
                         ComponentUtility.CopyComponent(cFromArr[i]);
                         ComponentUtility.PasteComponentAsNew(cToObj);
 
+                        if (adjustScale)
+                        {
+                            float mul = Helpers.GetScaleMultiplier(cFromArr[i].transform, cToObj.transform);
+                            if (cToObj.TryGetComponent(out SphereCollider sphere))
```
Comment (rurre) [inline Scripts/Editor/Copiers/LegacyCopier.cs:401]: What is TryGetComponent? Is this extension from the VRChat SDK?
The tool is supposed to be independent from the sdk so don't use it.

Try something like this instead:

```cs
Collider col = cToObj.GetComponent<Collider>();
if(col is BoxCollider box)
{
	box.center *= mul;
	box.size *= mul;
}
```

**Review by rurre — CHANGES_REQUESTED** [review]

File: Scripts/Editor/Copiers/LegacyCopier.cs (hunk 1 above)
Comment (Float3) [reply Scripts/Editor/Copiers/LegacyCopier.cs:401]: https://docs.unity3d.com/2019.4/Documentation/ScriptReference/Component.TryGetComponent.html is not from the VRChat SDK

Comment (Float3) [reply Scripts/Editor/Copiers/LegacyCopier.cs:401]:
```
    public unsafe bool TryGetComponent<T>(out T component)
    {
      CastHelper<T> castHelper = new CastHelper<T>();
      this.TryGetComponentFastPath(typeof (T), new IntPtr((void*) &castHelper.onePointerFurtherThanT));
      component = castHelper.t;
      return (object) castHelper.t != null;
    }
```

Comment (rurre) [reply Scripts/Editor/Copiers/LegacyCopier.cs:401]: Ah so it's new in 2019.4. Cool

**Pull request merged**